docs/src/docs/asciidoc/end-to-end-gradle-guide.adoc

```diff
@@ -0,0 +1,871 @@
+
+[source,json,subs="verbatim,attributes", role="multi-language-sample"]
+----
+{
+  "reflection": [
+    {
+      "condition": {
+        "typeReached": "org.example.NativeTests"
+      },
+      "type": "org.example.NativeTests$Person",
+      "allDeclaredFields": true
+    },
+    {
+      "condition": {
+        "typeReached": "sun.security.jca.GetInstance"
+      },
+      "type": "sun.security.provider.SHA",
+      "methods": [
+        {
+          "name": "<init>",
+          "parameterTypes": []
+        }
+      ]
+    }
+  ],
+  "resources": [
+    {
+      "condition": {
+        "typeReached": "org.example.NativeTests"
+      },
+      "glob": "TestResource.txt"
+    }
+  ],
+  "bundles": []
+}
+----
+
+As you can see there are no more entries that contain classes from `org.junit` (as their condition) for example.
+
+
+[[maintain-generated-metadata]]
+== Maintain generated metadata
+
+If you are a library maintainer, or your application became huge, you may consider covering most of your code with tests.
+This way you can also track if your project requires updates of the existing metadata.
+Considering that you run your tests in certain CI, at some point (after updating some dependency or adding new feature/test) you may notice some test failures with missing resources errors or that something is reflectively missing...
+In that case, your metadata requires update.
+
+[WARNING]
+Be aware that if you modified existing metadata file(s) on the default location, generating a new metadata will overwrite the existing one and your manual changes will be lost.
+
+So if you modified existing metadata file(s) on the default location, please do the following:
+
+1. Move your metadata to some other, non-default location (with the `metadataCopy` task for example)
+2. Set `mergeWithExisting` option to true in the `metadataCopy` block
+3. Run your tests again to generate new metadata (as we already described in the <<collect-metadata, collect metadata section>>)
+4. Run `metadataCopy` task again (with `mergeWithExisting` set to true in step 2)
+
+This way you will keep your original metadata, and add a new one.
+
+
+[[reachability-metadata-repository]]
+== Reachability metadata repository
+
+Native Build Tools (both Gradle and Maven plugins) picks metadata from Reachability metadata repository to ensure your application works out-of-box (if all metadata required by your app is already contributed to the metadata repository).
+Furthermore, you can configure Reachability metadata support through `metadataRepository` block added to our main `graalvmNative` block inside `build.gradle`.
+Most common options you may want to configure in this block are:
+
+* `enabled` - determines if you want to use Reachability metadata support or not (`true` by default)
+* `version` - specifies exact Reachability metadata version you want to use
+* `uri` - specifies the url where the metadata is stored. This can be used to point to the local repository
```

Comment:
```suggestion
* `uri` - specifies the url where the metadata is stored. This can be used to point to the local repository. If not set, it defaults to metadata which is shipped inside the Native Build Tools.
```